Examples/Tests/ionization_dsmc/analysis_ionization_dsmc_3d.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+# DSMC ionization test script:
+#   - compares WarpX simulation results with theoretical model predictions.
+
+import matplotlib.pyplot as plt
+import numpy as np
+import tqdm
+from openpmd_viewer import OpenPMDTimeSeries
+from scipy.stats import qmc
+
+from pywarpx import picmi
+
+constants = picmi.constants
+
+ts = OpenPMDTimeSeries("diags/diag2/")
+
+q_e = constants.q_e
+m_p = constants.m_p
+m_e = constants.m_e
+k_B = constants.kb
+ep0 = constants.ep0
+clight = constants.c
+
+plasma_density = 1e14
+neutral_density = 1e20
+dt = 1e-9
+electron_temp = 10
+neutral_temp = 300
+max_steps = 250
+max_time = max_steps * dt
+
+L = [0.1] * 3
+
+sigma_iz_file = "../../../../warpx-data/MCC_cross_sections/Xe/ionization.dat"
+iz_data = np.loadtxt(sigma_iz_file)
+
+energy_eV = iz_data[:, 0]
+sigma_m2 = iz_data[:, 1]
+iz_energy = energy_eV[0]
+
+
+def get_Te(ts):
+    T_e = []
+    for iteration in tqdm.tqdm(ts.iterations):
+        ux, uy, uz = ts.get_particle(
+            ["ux", "uy", "uz"], species="electrons", iteration=iteration
+        )
+        v_std_x = np.std(ux * clight)
+        v_std_y = np.std(uy * clight)
+        v_std_z = np.std(uz * clight)
+        v_std = (v_std_x + v_std_y + v_std_z) / 3
+        T_e.append(m_e * v_std**2 / q_e)
+    return T_e
+
+
+def get_density(ts):
+    number_data = np.genfromtxt("diags/counts.txt")
+    Te = get_Te(ts)
+    total_volume = L[0] * L[1] * L[2]
+    electron_weight = number_data[:, 8]
+    neutral_weight = number_data[:, 9]
+    ne = electron_weight / total_volume
+    nn = neutral_weight / total_volume
+    return [ne, nn, ne * Te]
+
+
+def compute_rate_coefficients(temperatures_eV, energy_eV, sigma_m2, num_samples=1024):
+    # """Integrate cross sections over maxwellian VDF to obtain reaction rate coefficients
+    # Given electron energy in eV (`energy_eV`) and reaction cross sections at those energies (`sigma_m2`),
+    # this function computes the reaction rate coefficient $k(T_e)$ for maxwellian electrons at
+    # a provided list of electron temperatures `temperatures_eV`.
+
+    # The rate coefficient is given by
+
+    # $$
+    # k(T_e) = \int \sigma(E) E dE
+    # $$
+
+    # where the energy $E$ is drawn from a maxwellian distribution function with zero speed and temperature $T_e$.
+    # We solve this using a quasi-monte carlo approach, by drawing a large number of low-discrepancy samples from
+    # the appropriate distribution and obtaining the average of $\sigma(E) E$.
+    # """
```

Comment:
No need for the `#` character in the docstring.
```suggestion
    """Integrate cross sections over maxwellian VDF to obtain reaction rate coefficients
    Given electron energy in eV (`energy_eV`) and reaction cross sections at those energies (`sigma_m2`),
    this function computes the reaction rate coefficient $k(T_e)$ for maxwellian electrons at
    a provided list of electron temperatures `temperatures_eV`.

    The rate coefficient is given by

    $$
    k(T_e) = \int \sigma(E) E dE
    $$

    where the energy $E$ is drawn from a maxwellian distribution function with zero speed and temperature $T_e$.
    We solve this using a quasi-monte carlo approach, by drawing a large number of low-discrepancy samples from
    the appropriate distribution and obtaining the average of $\sigma(E) E$.
    """
```